numpy/lib/twodim_base.py

```diff
@@ -573,7 +573,7 @@ def _histogram2d_dispatcher(x, y, bins=None, range=None, normed=None,
         N = len(bins)
     except TypeError:
         N = 1
-    if N != 1 and N != 2:
+    if N == 2:
```

Comment:
Let me suggest an alternative fix:
```python
    if N != 1 and N != 2:
        yield bins
    else:
        yield from bins  # bins=[x, y]
```

I like this better because the logic still matches up line for line with the function, which should make this easier to modify in the future.

Reply:
Looked at this again, and I think my implementation is right: for the `N=1` case we should just `yield bins` rather than `from bins`, since bins could very well just be a single integer (i.e., not iterable); only for a 2-element iterable does it contain separate items that could be overwritten).